Sources/LiveKit/Core/Engine.swift

```diff
@@ -211,23 +211,52 @@ class Engine: MulticastDelegate<EngineDelegate> {
             return
         }
 
+        guard publisher.iceConnectionState != .checking else {
```

Comment:
Is this used only for data channels? If so, we probably shouldn't re-negotiate if it's already connected.

Reply:
In the previous lines there is logic to prevent re-negotiation if already connected 🙂